File: 657-robot-return-to-origin/657-robot-return-to-origin.java
class Solution {
    public boolean judgeCircle(String moves) {
        int straight = 0, sideways = 0;
        for(int i = 0; i < moves.length(); i++){
            char move = moves.charAt(i);
            switch(move){
                case 'U':
                    straight++;break;
                case 'D':
                    straight--;break;
                case 'L':
                    sideways++;break;
                case 'R':
                    sideways--;break;
            }
        }
        return straight == 0 && sideways == 0 ;
    }
}

File: 657-robot-return-to-origin/657-robot-return-to-origin.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        straight = 0
        sideways = 0
        for move in moves:
            if move == "U": straight+=1
            if move == "D": straight-=1
            if move == "L": sideways+=1
            if move == "R": sideways-=1
        return straight==0 and sideways==0
                